### PR TITLE
Add BLS Aggregation window waiting after quorum

### PR DESCRIPTION
**Pull request by TomasArrachea**
Add BLS aggregation window-waiting-after-quorum feature. Once the aggregator reaches quorum, instead of sending the aggregated signature response, it will wait for a `window_duration` time. During that window time, it will continue to receive and aggregate new signatures, allowing to reduce the gas cost of validating the aggregation on chain. When the window time finishes, the aggregation is sent.
In the case that task's expiry time is reached during that window, the aggregation response is sent.

To implement this feature, the aggregation loop was modified. There are 3 posible cases:
- receiving a new signature,
- expiration timer,
- window timer (implemented using a channel and a tokio task).

Each time a new signature is received, it checks if quorum is reached. Once it's reached, the window timer fires. The loop continues until one of the two timers finish (window timer or task expiration).

This tests cases were added:
- signatures are processed during window after quorum
- if quorum has been reached and the task expires during window, the response is sent
- if window duration is zero, no signatures are aggregated after reaching quorum
- no signatures are aggregated after window

### CODE REVIEW

File: crates/services/bls_aggregation/src/bls_agg.rs [hunk 1]
```diff
@@ -303,117 +342,177 @@ impl<A: AvsRegistryService + Send + Sync + Clone + 'static> BlsAggregatorService
...
+                    let verification_failed = verification_result.is_err();
+
+                    digest
+                        .signature_verification_channel
+                        .send(verification_result)
+                        .await
+                        .map_err(|_| BlsAggregationServiceError::ChannelError)?;
+
+                    if verification_failed {
+                        continue;
```
Comment (supernovahs) [inline crates/services/bls_aggregation/src/bls_agg.rs:444]: can we add a debug! log here with appropriate params ?

File: crates/services/bls_aggregation/src/bls_agg.rs [hunk 2]
```diff
@@ -303,117 +342,177 @@ impl<A: AvsRegistryService + Send + Sync + Clone + 'static> BlsAggregatorService
...
+                    );
+
+                    if !BlsAggregatorService::<A>::check_if_stake_thresholds_met(
+                        &digest_aggregated_operators.signers_total_stake_per_quorum,
+                        &total_stake_per_quorum,
+                        &quorum_threshold_percentage_map,
+                    ) {
+                        continue;
+                    }
+
```
Comment (supernovahs) [inline crates/services/bls_aggregation/src/bls_agg.rs:491]: can we add debug! here," stake_thresholds_met " ?

File: crates/services/bls_aggregation/src/bls_agg.rs [hunk 3]
```diff
@@ -303,117 +342,177 @@ impl<A: AvsRegistryService + Send + Sync + Clone + 'static> BlsAggregatorService
...
+                    if !open_window {
+                        open_window = true;
```
Comment (supernovahs) [inline crates/services/bls_aggregation/src/bls_agg.rs:493]: debug! "opened the window" ?

File: crates/services/bls_aggregation/src/bls_agg.rs [hunk 4]
```diff
@@ -303,117 +342,177 @@ impl<A: AvsRegistryService + Send + Sync + Clone + 'static> BlsAggregatorService
...
+        let mut open_window = false;
+        let mut current_aggregated_response: Option<BlsAggregationServiceResponse> = None;
+        let (window_tx, mut window_rx) = tokio::sync::mpsc::unbounded_channel::<bool>();
+        let task_expired_timer = tokio::time::sleep(time_to_expiry);
+        tokio::pin!(task_expired_timer);
+
+        loop {
+            tokio::select! {
+                _ = &mut task_expired_timer => {
+                    // Task expired. If window is open, send aggregated reponse. Else, send error
```
Comment (supernovahs) [inline crates/services/bls_aggregation/src/bls_agg.rs:389]: debug! "task expired" ?

File: crates/services/bls_aggregation/src/bls_agg.rs (hunk 1 above)
Comment (TomasArrachea) [reply crates/services/bls_aggregation/src/bls_agg.rs:448]: Done!

**Review by supernovahs — APPROVED** [review]

**Pull request merged**